Documentation/INTERNALS.md

```diff
@@ -451,3 +451,94 @@ TODO
 
 TODO
 
+# Releasing
```

Comment:
I think it might make more sense to have this as a separate RELEASE.md document.